testing/functional/tests/directLine.test.js

```diff
@@ -12,25 +12,31 @@ const userMessage = 'Contoso';
 const directLineSecret = process.env.DIRECT_LINE_KEY || null;
 
 const auths = {
-    AuthorizationBotConnector: new Swagger.ApiKeyAuthorization(
-        'Authorization',
-        'BotConnector ' + directLineSecret,
-        'header'
-    ),
+    type: 'apiKey',
+    in: 'header',
+    name: 'Authorization',
+    value: 'BotConnector ' + directLineSecret,
 };
 
-function getDirectLineClient() {
-    return new Swagger({
+async function getDirectLineClient() {
+    const client = await Swagger({
         spec: directLineSpec,
         usePromise: true,
-        authorizations: auths,
+        requestInterceptor: (req) => {
+            if (auths && auths.in === 'header') {
+                req.headers[auths.name] = auths.value;
+            }
+            return req;
+        },
     });
+
+    return client;
 }
 
 async function sendMessage(client, conversationId) {
     let status;
     do {
-        await client.Conversations.Conversations_PostMessage({
+        return client.apis.Conversations.Conversations_PostMessage({
```

Comment:
This method is not async anymore?

Reply:
If we are returning instead of awaiting, i think the do while will not take effect, it will just return it the first time, right?